### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by victor-rds**
* Update to requirments.txt based on the same file used of etesync/journal-manager;
* Update etesync/journal-manager to v0.5.2
* Update links on config files to Django 2.0
* Update urls.py to the new Django syntax

### CODE REVIEW

File: etesync_server/urls.py [hunk 1]
```diff
@@ -30,11 +30,10 @@
...
-    url(r'^api/v1/', include(router.urls)),
-    url(r'^api/v1/', include(journals_router.urls)),
-    url(r'^api-auth/', include('rest_framework.urls', namespace='rest_framework')),
-    url(r'^api-token-auth/', token_views.obtain_auth_token),
-    url(r'^admin/', admin.site.urls),
+    re_path(r'^api/v1/', include(router.urls)),
+    re_path(r'^api/v1/', include(journals_router.urls)),
+    re_path(r'^api-auth/', include('rest_framework.urls', namespace='rest_framework')),
+    re_path(r'^api-token-auth/', token_views.obtain_auth_token),
+	path('admin/', admin.site.urls),
```
Comment (tasn) [inline etesync_server/urls.py]: Indentation error.

File: etesync_server/settings.py [hunk 2]
```diff
@@ -16,21 +16,18 @@
...
 # Quick-start development settings - unsuitable for production
-# See https://docs.djangoproject.com/en/1.10/howto/deployment/checklist/
+# See https://docs.djangoproject.com/en/2.0/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
-# See secret.py for how this is generated; uses a file 'secret.txt' in the root
-# directory
-SECRET_KEY = get_secret_from_file(os.path.join(BASE_DIR, "secret.txt"))
+# See secret.py for how this is generated; uses a file 'secret.txt' in the root directory
+SECRET_KEY = get_secret_from_file(os.environ.get('ETESYNC_SECRET_FILE',os.path.join(BASE_DIR, 'secret.txt')))
```
Comment (tasn) [inline etesync_server/settings.py]: Unrelated change. This actually looks great, and I'd love to have it, but it's unrelated to this PR. Could you please split this to a different PR, or at the very least, a different commit?

File: etesync_server/settings.py [hunk 3]
```diff
@@ -122,11 +118,10 @@
...
+# https://docs.djangoproject.com/en/2.0/howto/static-files/
 
 STATIC_URL = '/static/'
 
-
 JOURNAL_API_PERMISSIONS = (
-        'rest_framework.permissions.IsAuthenticated',
-        )
+	'rest_framework.permissions.IsAuthenticated',
+)
```
Comment (tasn) [inline etesync_server/settings.py]: Please avoid changing formatting and logic in the same commit.

File: requirements.txt [hunk 4]
```diff
@@ -1,5 +1,5 @@
-Django>=1.11,<1.11.999
-djangorestframework>=3.7,<3.7.999
-drf-nested-routers==0.90.0
+Django
+djangorestframework
+drf-nested-routers
```
Comment (tasn) [inline requirements.txt]: This change is unrelated to the overall change, and is probably unwanted. Limiting the range of allowed versions makes sure that future breaking changes don't break installations. Could you please just update the ranges to the updated ones?

Comment (victor-rds) [reply requirements.txt]: Ok, in this specific case I used the `requirements.txt` from the journal-mapper repository for the up to date dependencies, but I see your point.

File: etesync_server/settings.py (hunk 2 above)
Comment (victor-rds) [reply etesync_server/settings.py]: Ops! That one is from my docker image settings file!

File: requirements.txt (hunk 4 above)
Comment (tasn) [reply requirements.txt]: It's a slightly different usecase. There we want to encourage people to use the latest, so when they develop, they always test against latest. Here we are trying to create stability.

**Comment by tasn** [comment]
Merged, thanks!
In the future though, please split unrelated changes to different commits.

\o/
